### ISSUE TITLE
[Feature Request] Time-line thumbnail preview.

### ISSUE THREAD

**Issue by LichenSymbiont** [mpv-player/mpv#6919]
It would be an exceedingly impressive feature for such a light-weight media player!

But at the moment, it's only possible through the library with taking an image from the video, shrinking it and displaying it. Perfectly simple, but performance-to-usefulness ratio may not be sufficient -- developers decide.

If this can be done in a genius way, well, then let's get it done already!

**Comment by ksharindam** [comment]
There is a plugin for it.
[mpv thumbnail preview script](https://github.com/TheAMM/mpv_thumbnail_script)

**Comment by oakkitten** [comment]
the script linked works, but it's got a number of disadvantages:

  * it's slow, taking minutes to run even on okay hardware
  * uses a lot of cpu
  * low time resolution for long videos (configurable, but then see the above points)
  * doesn't really work with files currently being downloaded
  * leaves thumbnails behind

for the developer of bomi, which is itself based on mpv, it took [one day to implement](https://github.com/xylosper/bomi/commit/098af478ddec7fead50599c29299a1c3fe76ead3) this feature, and it's bullet fast and has none of these issues.

i really hope mpv implements this. scripts are good but sometimes scripts just don't cut it

**Comment by zc62** [comment]
The thing is, mpv does not really have a GUI. The pseudo-gui is implemented by a lua script. And the above script just replaces the built-in OSC. I doubt if one can do better than that script if he is only modifying [`osc.lua`](https://github.com/mpv-player/mpv/blob/master/player/lua/osc.lua). AFAIK mpv does not expose image raw data to scripts.

**Comment by oakkitten** [comment]
i guess all we need is something like a `mp.get_thumbnail(timestamp)` lua call, then?

**Comment by stax76** [comment]
I also think this should be built into the default OSC.

**Comment by wm4** [comment]
Why does the thumbnail script not suffice to you?

I've always played with the thought of providing scripts a way to extract decoded frames from demuxer-cached regions, but not outside. While this would be pretty efficient and (mostly) work with web streams, I'm not sure if it'd make users happy. Probably not, so all effort would be in vain.

> other players have this feature implemented without these problems. for the developer of bomi, which is itself based on mpv, it took one day to implement it!

This starts a separate libmpv instance. That means it opens the file again. It pauses playback, disables audio, and lets the libmpv render into a tiny rectangle (that's your thumbnail). mpv could do exactly the same, but not sure if I want it to.

**Comment by v-fox** [comment]
> Why does the thumbnail script not suffice to you?

@wm4 Seeing bomi's glorious seek-bar, which is the greatest thing that ever happened to video players (just like Amarok/Clementine's [moodbar](https://en.wikipedia.org/wiki/Moodbar) for audio-players), would be the best self-explanation why external script unlikely to compare. Especially when you open some 10-100GB ≥1080p video file and get instant preview picture on whole timeline with second-precision, even as big as ≥640x480.

> This starts a separate libmpv instance. That means it opens the file again. It pauses playback, disables audio, and lets the libmpv render into a tiny rectangle (that's your thumbnail). mpv could do exactly the same, but not sure if I want it to.

I guess, you would still have to run another, hidden player instance and have some lower-quality profile to render it fast (seek-time may really suck on primary profile) but not distorted. And it will not play well with mpv's simple UI, unless it renders it as overlay right in the windows of "primary" instance, unlike bomi's preview overlay window.

**Comment by mindfulsource** [comment]
Just wanted to comment, IINA (for mac users) has this feature. So at least there's that solution for some of us. But  to work the file has to be local (not network file). When it does, it's very smooth, though the thumbnail is very small (at least by default).

**Comment by nvcken** [comment]
I tried to setup mpv thumbnail preview script, but doesn't work yet.
https://github.com/TheAMM/mpv_thumbnail_script/issues/61
Have anyone tried it? Pls help

**Comment by stax76** [comment]
Yes I think I had tried it and a user of mpv.net but long time ago, it was working.
